File: Sample/CSharp/ScorpioProto/Commons/IScorpioReader.cs
using System;
namespace ScorpioProto.Commons {
    public interface IScorpioReader {
        bool ReadBool();
        sbyte ReadInt8();
        byte ReadUInt8();
        short ReadInt16();
        ushort ReadUInt16();
        int ReadInt32();
        uint ReadUInt32();
        long ReadInt64();
        ulong ReadUInt64();
        float ReadFloat();
        double ReadDouble();
        string ReadString();
        DateTime ReadDateTime();
        byte[] ReadBytes();
    }
}

File: Backup/Typescript/src/ScorpioProto/Commons/IScorpioReader.ts
export interface IScorpioReader {
    ReadBool():boolean;
    ReadInt8():number;
    ReadInt16():number;
    ReadInt32():number;
    ReadInt64():any;
    ReadFloat():number;
    ReadDouble():number;
    ReadString():string;
    ReadDateTime():Date;
}